js/bybit.js

```diff
@@ -4000,36 +4314,69 @@ module.exports = class bybit extends Exchange {
          */
-        const rawNetwork = this.safeStringUpper (params, 'network');
-        const networks = this.safeValue (this.options, 'networks', {});
-        const network = this.safeString (networks, rawNetwork, rawNetwork);
-        params = this.omit (params, 'network');
-        const response = await this.fetchDepositAddressesByNetwork (code, params);
-        let result = undefined;
-        if (network === undefined) {
-            result = this.safeValue (response, code);
-            if (result === undefined) {
-                const alias = this.safeString (networks, code, code);
-                result = this.safeValue (response, alias);
+        const [ networkId, query ] = this.handleNetworkCodeAndParams (params);
+        if (networkId !== undefined) {
+            const currency = this.currency (code);
+            const request = {
+                'coin': currency['id'],
+                'chainType': networkId,
+            };
+            const response = await this.privateGetAssetV3PrivateDepositAddressQuery (this.extend (request, query));
+            //
+            //    {
+            //         "retCode": "0",
+            //         "retMsg": "success",
+            //         "result": {
+            //             "coin": "USDT",
+            //             "chains": [
+            //                 {
+            //                     "chainType": "TRC20",
+            //                     "addressDeposit": "TC6NCAC5WSVCCiaD3kWZXyW91ZKKhLm53b",
+            //                     "tagDeposit": "",
+            //                     "chain": "TRX"
+            //                 }
+            //             ]
+            //         },
+            //         "retExtInfo": {},
+            //         "time": "1666895654316"
+            //     }
+            //
+            const result = this.safeValue (response, 'result', {});
+            const chains = this.safeValue (result, 'chains', []);
+            const addressInfo = this.safeValue (chains, 0, {});
+            return this.parseDepositAddress (addressInfo, currency);
+        } else {
+            const rawNetwork = this.safeStringUpper (params, 'network');
+            const networks = this.safeValue (this.options, 'networks', {});
+            const network = this.safeString (networks, rawNetwork, rawNetwork);
+            params = this.omit (params, 'network');
+            const response = await this.fetchDepositAddressesByNetwork (code, query);
+            let result = undefined;
+            if (network === undefined) {
+                result = this.safeValue (response, code);
                 if (result === undefined) {
-                    const defaultNetwork = this.safeString (this.options, 'defaultNetwork', 'ERC20');
-                    result = this.safeValue (response, defaultNetwork);
+                    const alias = this.safeString (networks, code, code);
```

Comment:
4 nested ifs it's just bad practice and makes the code hard to read, are you sure we can't replace this with a loop or something?

Reply:
Well, if you look at that method, I've not added those nested if/else in this PR, and I've just added only 1 `if` in the top. so, if possible to ignore this fact atm, because of the complexity of the merge of this PR, let's put that if/else nesting problem aside, and after my & peter's PRs are merged, I'll then make reorganize of this part. Does that sound ok?